src/main/java/org/broadinstitute/hellbender/tools/copynumber/allelic/alleliccount/AllelicCount.java

```diff
@@ -18,30 +18,29 @@
     private final SimpleInterval interval;
     private final int refReadCount;
     private final int altReadCount;
-    /* these are extra metadata and can be null */
+    /* these are set to N if not specified */
     private final Nucleotide refNucleotide;
     private final Nucleotide altNucleotide;
 
     /**
      * Construct the allelic count object.
      * @param interval the genomic interval (it is assumed to have the same begin and end points)
-     * @param refReadCount ref nucleotide count in the pileup
-     * @param altReadCount alt nucleotide count in the pileup
+     * @param refReadCount ref nucleotide count
+     * @param altReadCount alt nucleotide count
      * @param refNucleotide ref nucleotide
      * @param altNucleotide alt nucleotide
      */
     public AllelicCount(final SimpleInterval interval, final int refReadCount, final int altReadCount,
                         final Nucleotide refNucleotide, final Nucleotide altNucleotide) {
-        this.interval = Utils.nonNull(interval, "Can't construct AllelicCount with null interval.");
+        this.interval = Utils.nonNull(interval);
         this.refReadCount = ParamUtils.isPositiveOrZero(refReadCount, "Can't construct AllelicCount with negative read counts.");
         this.altReadCount = ParamUtils.isPositiveOrZero(altReadCount, "Can't construct AllelicCount with negative read counts.");
-        /* these can be null */
-        this.refNucleotide = refNucleotide;
-        this.altNucleotide = altNucleotide;
+        this.refNucleotide = Utils.nonNull(refNucleotide);
+        this.altNucleotide = Utils.nonNull(altNucleotide);
     }
 
     public AllelicCount(final SimpleInterval interval, final int refReadCount, final int altReadCount) {
-        this(interval, refReadCount, altReadCount, null, null);
+        this(interval, refReadCount, altReadCount, Nucleotide.N, Nucleotide.N);
```

Comment:
Do we ever need this constructor?

Reply:
Originally, `AllelicCount` didn't even store the nucleotides, since the allele-fraction only needs the count information.  @mbabadi added the nucleotides for GetBayesianHetCoverage.  So once we port over a v1.5 allele-fraction model to the new copynumber package, I expect that this constructor could come in handy.